e2e_test/s3/fs_source_v2_new_file.py

```diff
@@ -71,6 +71,8 @@ def _table():
         config, len(data_batch1), ITEM_NUM_PER_FILE, run_id, fmt, need_drop_table=False
     )
 
+    time.sleep(10)
+
```

Comment:
Can we make this test more robust? e.g., sth like: 1. upload some files. 2. check. 3. upload another batch of files. 4. check again

Reply:
I think it does the same thing

first check https://github.com/risingwavelabs/risingwave/blob/d5f503e15663beab4e73005a47ef08bfca67a740/e2e_test/s3/fs_source_v2_new_file.py#L70 

check after upload
https://github.com/risingwavelabs/risingwave/blob/d5f503e15663beab4e73005a47ef08bfca67a740/e2e_test/s3/fs_source_v2_new_file.py#L79

Reply:
Then why it didn't catch the bug before? :eyes:

Reply:
I am surprised, too. 😄

Reply:
So can we make this test more robust like doing **interleaving** upload and check? If a single `sleep` can affect the correctness, it seems too fragile.

Reply:
Sorry, I misunderstood it. `do_test` is the first check, and `check_for_new_files` is the second check. I just looked for the usage of `check_for_new_files` and thought it's the only check.